docs/assignment3.md

```diff
@@ -1,110 +1,776 @@
+ 
+The second one is when the user wants to change the color of the interface. The color of choice in this situation is “The Pink''. First the application is at its initial state – the dashboard. From there, after pressing the “Settings”' button, it goes to a new class, performing a main action changing the interface into the page of the settings’ page. In order to go to the settings’ page, the program goes through a class called ApplicationState, redirecting it to the SettingsForm class. From there, the user has the options of doing three sub actions – change the default programming language of the snippet helper, change the color theme of the interface and to add some new categories into the database of the application. From then on the user would choose to change the color. A falling menu shows when the user presses the bar that enables this action. Three new sub actions are available for the user – to change the color to green, orange or pink. In this situation the user chooses the color Pink. Then the program goes through a switch case taking the input of the user and goes to another class – Settings – changing the color of the theme as it returns the value of “pink” to the SettingsForm class. The new color would appear when the user refreshes the page or, as a whole, perform a main action from the menu that enables the user to change the interface at any point. 
+
+`Figure representing the Sequence Diagram 2`\
+![Image of Sequence Diagram 2](./diagrams/assignment3/seq_d_assginment3_scenario2.png)
+ 
+ 
+The third one is when the user wants to search for a snippet with a non-existent name. We start from the Dashboard class. The user presses the button for entering the All Snippets main action. Before changing the interface of the page to the one of the one we want to enter, the system goes through the `ApplicationState` class which then goes to the `AllSnippets` class. In that state, the system call the function `AllSnipetsForm`, in order for the user to access the list of snippets and presents the options of doing some sub actions like filtering the list of snippets by their date of creation – ascending and descending, sorting the list by the language that they are written on and allowing choosing more than 1 language, sorting the languages by their name as well. In this case, we will be looking at the sub action looking for a snippet by the name that the snippet was saved with. The system takes the input of the user and goes into a loop checking the already existing snippets and as the loop goes, the bar showing the snippets is getting smaller as the bar with the name gets more specific. Since the user is trying to find a snippet with a name that does not exist in the system beforehand, the bar with the supposed result becomes empty. 
+
+`Figure representing the Sequence Diagram 3`\
+![Image of Sequence Diagram 3](./diagrams/assignment3/seq_d_assginment3_scenario3.png)
+
```

Comment:
Good overall sequence diagrams and descriptions. Missing the min/max number of iterations in the "loop" fragment. You should've used the alt fragment instead of calling it 'switch'. Make sure to use the 'strict' fragment when using atomic sequences. Different names for the objects in your sequence diagram as your instances of classes present in the class diagram, they should be aligned. The sequences of messages seem to be aligned with your java implementation, good job. Overall, good descriptions and model represents most things needed. Just some mistakes with the usage of UML constructs.